### PR TITLE
StackEditor: Credentials on new SideView

### PR DESCRIPTION
**Pull request by gokmen**
Currently only lists existing credentials, updates in use states based on the selected stack template  and provides preview/edit/delete functionality over them. 

Edit functionality will be moved into it's view while new credential create flow implementation.

### CODE REVIEW

**Comment by gokmen** [comment]
@usirin this PR also includes fixes for previous PR https://github.com/koding/koding/pull/10181 fyi.

**Comment by gokmen** [comment]
cc/ @rjeczalik we don't need to provide `config.requiredData` and `config.requiredProviders` anymore. It will be parsed and updated in backend.

File: client/new-stack-editor/lib/styl/main.styl [hunk 1]
```diff
@@ -17,11 +17,15 @@ $dark-tab-color             = #5C5C5C
...
 $resize-handle-color        = #6EA3E5
 $stack-bg-color             = #232327
+$stack-bg-color-bright      = #29292D
 $stack-fg-color             = #59595C
 $stack-fg-color-bright      = #DDDDDD
 $stack-fg-color-yellow      = #FCC34E
 $stack-border-color         = #000000
 
+$stack-sideview-width       = 420px
+$stack-border-width         = 2px
```
Comment (usirin) [inline client/new-stack-editor/lib/styl/main.styl]: 👍

File: client/new-stack-editor/lib/views/credentiallistitem.coffee [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+  JView.mixin @prototype
+
+  constructor: (options = {}, data) ->
+
+    options.cssClass = kd.utils.curry 'credential', options.cssClass
+
+    super options, data
+
+    handle = (action) => =>
```
Comment (usirin) [inline client/new-stack-editor/lib/views/credentiallistitem.coffee]: This is gonna define a new handler for each credential list item, making it prototype method would be more efficient here, as it's gonna be defined and reused each time.

File: workers/social/lib/social/clientrequire.coffee [hunk 3]
```diff
@@ -0,0 +1,2 @@
+module.exports = clientRequire = (path) ->
+  require "#{process.env.KONFIG_PROJECTROOT}/client/#{path}"
```
Comment (usirin) [inline workers/social/lib/social/clientrequire.coffee]: 👍

File: workers/social/lib/social/models/computeproviders/stacktemplate.coffee [hunk 4]
```diff
@@ -433,18 +449,25 @@ module.exports = class JStackTemplate extends Module
       delete data.group
 
       # Update template sum if template update requested
-      { template, templateDetails, rawContent } = data
+      { config, template, templateDetails, rawContent } = data
+      config ?= {}
```
Comment (usirin) [inline workers/social/lib/social/models/computeproviders/stacktemplate.coffee]: [minor] default args on object destructuring works like argument destructuring:

```coffee
{ config, template, templateDetails, rawContent } = data
config ?= {}
```
vs
```coffee
{ config = {}, template, templateDetails, rawContent } = data
```

**Review by usirin — APPROVED** [review]
LGTM 👍 only minor parts, but can be tackled later.

File: client/new-stack-editor/lib/views/credentiallistitem.coffee (hunk 2 above)
Comment (gokmen) [inline client/new-stack-editor/lib/views/credentiallistitem.coffee]: right 👍  tought the same was lazy to change :) will be updated in next iteration.

File: workers/social/lib/social/models/computeproviders/stacktemplate.coffee (hunk 4 above)
Comment (gokmen) [inline workers/social/lib/social/models/computeproviders/stacktemplate.coffee]: TIL 🥇

**Comment by rjeczalik** [comment]
> we don't need to provide config.requiredData and config.requiredProviders anymore. It will be parsed and updated in backend.

What about title? (https://github.com/koding/koding/pull/9488/files#r95247360)

**Comment by gokmen** [comment]
@rjeczalik I can provide defaults for those as well 👍  I'll update you once I'm done then you can remove the implementation on your side.

**Pull request merged**